### DISCUSSION TITLE
Работа со значением из префильтра

### DISCUSSION THREAD

**Discussion by vldekov** [totumonline/totum-ru-issues-and-discussions#333] [Как это сделать?] [answered]
Как возможно взять и передать значение из префильтра в код действия кнопки?
=:linkToDataText(title: "Подразделение"; text: #fl_podrazdelenie_f)
результат: пусто

**Comment by ttm-alexeypolunin** [comment, marked as answer]
Решеткой. Но надо учитывать, что префильтр существует только для web-канала! В инннере его нет! Те если вы из другой таблицы выполните execButton этой кнопки то это значение будет пустым.

**Reply by ttm-alexeypolunin** [reply]
В вашем случае возможно в префильтре с подразделением лежит список (например это мультиселект), а массивы они перпендикулярны тексту (так можно представить). Если там массив, то его надо в строку/текст превратить. Для списка например склеив через listJoin

**Reply by vldekov** [reply]
поле Подразделение в префильтре и строчной части это строка
с решеткой результат пусто
![image](https://user-images.githubusercontent.com/114401808/224941730-6b3934ce-d332-47fa-8b80-2f3ffdfc93b0.png)
необходимо подписать поле Бригадир с помощью setList

**Reply by ttm-alexeypolunin** [reply]
А, кнопка в префильтре - мы ее недавно туда добавили, там может что-то и не работать. Посмотрим сег.

**Reply by vldekov** [reply]
такая же проблема и с кнопкой в хедере
высылаю код, может проблема в том что использую код тотум ?

**Reply by vldekov** [reply]
```
= :linkToInputSelect(title: "Выберите фамилию бригадира!"; code: $code1; codeselect: $code2; multiple: false; refresh: true; button: "Сохранить"; close: false)

    ```code1:totum
    =:setList(table: $#ntn; field: 'brigadir' = $#input;where: 'podrazdelenie'=#fl_podrazdelenie_f;  log: true)
    ```

    ```code2:totum
    =:selectRowListForSelect(table: 'tablicabrigadirov'; field: 'krtakoefio';bfield: 'krtakoefio')
    ```
````

**Reply by ttm-alexeypolunin** [reply]
**В общем мы проверили — с кнопкой все ок.**

Про что надо помнить, что значения в Префильтре появляются после нажатия на `GO`.

До этого момента они совсем виртуальные.

**Comment by vldekov** [comment]
нашел решение. проблема была в коде тотум.

```
= :linkToInputSelect(title: "Выберите фамилию бригадира!"; var: 'podrazdelenie_' = #fl_podrazdelenie_f; var: 'uchastok_'=#fl_uchastok_f; code: $code1; codeselect: $code2; multiple: false; refresh: true; button: "Сохранить"; close: false)

    ```code1:totum
    =:setList(table: $#ntn; field: 'brigadir' = $#input;where: 'podrazdelenie'=$#podrazdelenie_;where: 'uchastok'=$#uchastok_; ;  log: true)
    ```

    ```code2:totum
    =:selectRowListForSelect(table: 'tablicabrigadirov'; field: 'krtakoefio';bfield: 'krtakoefio')
    ```

```
